classes/submission/reviewAssignment/ReviewAssignmentDAO.inc.php

```diff
@@ -376,15 +375,21 @@ function insertObject($reviewAssignment) {
 		);
 
 		$reviewAssignment->setId($this->getInsertId());
-		return $reviewAssignment->getId();
+
+		// Update review stage status whenever a review assignment is changed
+		if ($result) {
```

Comment:
Database errors should result in something stronger than a `false` -- typically a fatal error. In the past, devs have misread the boolean return as an indication of whether any records matched, which isn't the case.